### PR TITLE
feat(topic_state_monitor): support transform topic check

### PR DESCRIPTION
**Pull request by isamu-takagi**
Signed-off-by: Takagi, Isamu <isamu.takagi@tier4.jp>

## Description

Add transform check mode to topic_state_monitor for https://github.com/autowarefoundation/autoware.universe/pull/1569.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: system/topic_state_monitor/Readme.md [hunk 1]
```diff
@@ -35,23 +35,25 @@ The types of topic status and corresponding diagnostic status are following.
...
-| `timeout`         | double | 1.0           | If the topic subscription is stopped for more than this time [s], the topic status becomes `Timeout` |
+| Name          | Type   | Default Value | Description                                                                                          |
+| ------------- | ------ | ------------- | ---------------------------------------------------------------------------------------------------- |
+| `warn_rate`   | double | 0.5           | If the topic rate is lower than this value, the topic status becomes `WarnRate`                      |
+| `error_rate`  | double | 0.1           | If the topic rate is lower than this value, the topic status becomes `ErrorRate`                     |
+| `timeout`     | double | 1.0           | If the topic subscription is stopped for more than this time [s], the topic status becomes `Timeout` |
+| `window_size` | int    | 10            | Window size of target topic for calculating frequency                                                |
 
 ## Assumptions / Known limits
```
Comment (kenji-miyake) [inline system/topic_state_monitor/Readme.md:59]: What made you change the categories of parameters?
I feel it should be done in another PR to minimize the diff of this PR.

Comment (isamu-takagi) [reply system/topic_state_monitor/Readme.md:59]: I have split the PR.
https://github.com/autowarefoundation/autoware.universe/pull/1586

Comment (kenji-miyake) [reply system/topic_state_monitor/Readme.md:59]: Thank you very much! :pray:

**Review by kenji-miyake — APPROVED** [review]
